base_layer/core/src/base_node/sync/header_sync/validator.rs

```diff
@@ -138,8 +138,7 @@ impl<B: BlockchainBackend + 'static> BlockHeaderSyncValidator<B> {
             // We dont want to mark a block as bad for internal failures
             Err(
                 e @ ValidationError::FatalStorageError(_) |
-                e @ ValidationError::IncorrectNumberOfTimestampsProvided { .. } |
-                e @ ValidationError::AsyncTaskFailed(_),
```

Comment:
Another logic change, are you sure it's correct?

Reply:
This is correct. 
This only suppresses internal errors to let them be the reason a block is flag as a bad block